tests/golos.vesting_tests.cpp

```diff
@@ -221,6 +221,27 @@ BOOST_FIXTURE_TEST_CASE(buy_vesting, golos_vesting_tester) try {
     CHECK_MATCHING_OBJECT(vest.get_stats(), mvo()("supply", vest.asset_str(buy1+buy2)));
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(bulk_buy_vesting, golos_vesting_tester) try {
+    BOOST_TEST_MESSAGE("Test bulk buying vesting / converting token to vesting");
+    BOOST_CHECK_EQUAL(success(), token.create(cfg::emission_name, token.make_asset(1000), {cfg::charge_name, cfg::publish_name}));
+    BOOST_CHECK_EQUAL(success(), token.issue(cfg::emission_name, N(sania), token.make_asset(500), "issue tokens sania"));
+    produce_block();
+
+    BOOST_CHECK_EQUAL(success(), vest.create_vesting(cfg::emission_name));
+    BOOST_CHECK_EQUAL(success(), vest.open(N(sania)));
+    BOOST_CHECK_EQUAL(success(), vest.open(N(pasha)));
+    BOOST_CHECK_EQUAL(success(), vest.open(N(tania)));
+    BOOST_CHECK_EQUAL(success(), vest.open(N(vania)));
+    produce_block();
+
+    BOOST_CHECK_EQUAL(success(), token.bulk_transfer( N(sania), {{cfg::vesting_name, token.make_asset(200), "buy vesting"},
+                                                                 {cfg::vesting_name, token.make_asset(100), "buy vesting"}}
+                                                    ));
+
+    CHECK_MATCHING_OBJECT(token.get_account(N(sania)), mvo()("balance", token.asset_str(200)));
+    CHECK_MATCHING_OBJECT(vest.get_balance(N(sania)), vest.make_balance(300));
```

Comment:
please add test of different assets in one bulktransfer . there should be at least 2 transfers to with same asset (to test sum works) and also it should test fail if asset not registered in vesting.
OR we can remove this functionality for now and add assert to cyber.token to disallow different assets in one bulktransfer

Reply:
this is implemented in the task https://github.com/GolosChain/cyberway.contracts/issues/161